gramex/data.py

```diff
@@ -1470,9 +1509,13 @@ def _mongodb_json(obj):
 def _filter_mongodb(url, controls, args, database=None, collection=None, query=None, **kwargs):
     '''TODO: Document function and usage'''
     table = _mongodb_collection(url, database, collection, **kwargs)
-    if query is None:
-        meta_cols = pd.DataFrame(list(table.find().limit(100)))
-        query = _mongodb_query(args, meta_cols)
+
+    results = table.find().limit(100)
+    meta_cols = pd.DataFrame(list(copy.deepcopy(results)))
```

Comment:
Check this out. `meta_cols` is being used only for

- `_controls_default` -- which just requires a list of column names
- `_filter_col`  -- which just requires a list of column names

So we don't need the DataFrame